task_arrays/js/main.js

```diff
@@ -0,0 +1,127 @@
+"use strict";
+
+var values = [[4], 0, undefined, 2, [36, "hg"], "2", null, false, null, 8, "rt"];
+
+values.forEach(logIsArray);
+var arrayRange = range(1, 10, 2);
+logConsole(arrayRange);
+logConsole(values, compactFor);
+logConsole(values, compactFilter);
+logConsole(arrayRange, sumFor);
+logConsole(arrayRange, sumReduce);
+logConsole(values, unique);
+logConsole(values, last);
+excludeLast(arrayRange, 2);
+logConsole(arrayRange);
+
+function logIsArray(value) {
+  console.log(value, ': ' + isArrayClass(value) + ' : ' + isArrayConstructor(value) + ' : ' +
+          isArrayInstanceof(value));
+}
+
+function logConsole(value, func) {
+  if (func) {
+    console.log(func.name + ":", func(value));
+  } else {
+    console.log(value);
+  }
+}
+
+function isArrayClass(arg) {
+  if (!isObject(arg)) {
+    return false;
+  }
+  return {}.toString.call(arg).slice(8, -1) === 'Array';
+}
+
+function isArrayConstructor(arg) {
+  if (!isObject(arg)) {
+    return false;
+  }
+  return arg.constructor.name === 'Array';
+}
+
+function isArrayInstanceof(arg) {
+  if (!isObject(arg)) {
+    return false;
+  }
+  return arg instanceof Array;
+}
+
+function isObject(arg) {
+  if (arg === null) {
+    return false;
+  }
+  return typeof arg === 'object';
+}
+
+function range() {
```

Comment:
next cases should work:
range(10) [0..9]
range(1, 10) [1..9]
range(1, 10, 3) [1, 4, 7]
range(10, null, 3) [0, 3, 6, 9]
range(10, null) the same with range(10)
range(-2, -5) []
range(-5, -2) [-5, -4, -3]
range(-5, -2, 2) [-5, -3]
range(-5, -2, -2) []
range(-5, null, 2) []
range(-5, null, -2) [0, -2, -4]
range(-10, -20, -5) [-10, -15]
range(-20, -10, -5) []

FYI this function can be implemented ~10 lines, firstly you should find step and length of result array (the difficult part of this function)